src/uint/inv_mod.rs

```diff
@@ -106,7 +106,10 @@ impl<const LIMBS: usize> Uint<LIMBS> {
 
         // Decompose `self` into RNS with moduli `2^k` and `s` and calculate the inverses.
         // Using the fact that `(z^{-1} mod (m1 * m2)) mod m1 == z^{-1} mod m1`
-        let maybe_a = self.inv_odd_mod(&Odd(s));
+        let s_is_odd = s.is_odd();
+        let mut maybe_a = self.inv_odd_mod(&Odd(s));
+        maybe_a.is_some = maybe_a.is_some.and(s_is_odd);
```

Comment:
This seems somewhat like https://doc.rust-lang.org/std/option/enum.Option.html#method.filter

Reply:
What I mean is, I would prefer it to be a method instead of mutating `maybe_a`

Reply:
Hmm yeah, something like that might be doable in a `const fn` context

Reply:
Oh maybe not... it would need an actual closure (as opposed to a function pointer) to be able to query `s_is_odd` from the outer scope and still be callable from a `const fn`.

Reply:
So really it would have to be a method that accepts a `ConstChoice` and applies an `&=` style mutation, but it could consume the original `ConstCtOption` and return a new one.

I guess the question is what to call such a method.

Reply:
To implement `filter`, yes, but we only need a kind of `flatten()` functionality. Some `fn ___(&self, is_some: ConstChoice) -> Self`. `some_if()`?

Reply:
[222685d](https://github.com/RustCrypto/crypto-bigint/pull/523/commits/222685d6e4d59c028eb5b0e054845ddd7c6842af) adds `ConstCtOption::and_choice` (for lack of a better name) to wrap up the mutation

Reply:
I'm open to better options for a name

Reply:
Maybe make it `pub(crate)` for now, so that it's easier to change later if we come up with something better